File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max_store = Integer.MIN_VALUE, max_sum = 0;

        for(int i = 0; i < nums.length; i++) {
            max_sum = max_sum + nums[i];

            if( max_sum > max_store ) max_store = max_sum;

            if( max_sum < 0 ) max_sum = 0;
        }
        
        return max_store;
    }
}

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        // kadane algorithm
        // -ve mila -> result 0;
        // +ve mila -> result += n;

        int maxSum = INT_MIN, currSum = 0;
        for (auto& n : nums) {
            currSum += n;
            maxSum = max(maxSum, currSum);

            if (currSum < 0) {
                currSum = 0;
            }
        }
        return maxSum;
    }
};